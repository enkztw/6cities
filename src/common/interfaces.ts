import {city, id, property} from './types';

export interface Location {
    latitude: number;
    longitude: number;
    zoom: number;
}

export interface City {
    location: Location;
    name: city;
}

export interface User {
    readonly id: id;
    avatarUrl: string;
    isPro: boolean;
    name: string;
    email?: string;
    description?: string;
}

export interface Review {
    readonly id: id;
    comment: string;
    date: string;
    rating: number;
    user: User;
}

export interface Status {
    isAuthorized: boolean;
}

export interface Offer {
    readonly id: id;
    bedrooms: number;
    city: City;
    description: string;
    goods: string[];
    host: User;
    images: string[];
    isFavorite: boolean;
    isPremium: boolean;
    location: Location;
    maxAdults: number;
    previewImage: string;
    price: number;
    rating: number;
    title: string;
    type: property;
}
