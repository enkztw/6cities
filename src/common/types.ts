export type property = `apartment` | `room` | `house` | `hotel`;
export type id = number
export type city = `Paris` | `Cologne` | `Brussels` | `Amsterdam` | `Hamburg` | `Dusseldorf`
export type sort = `rating` | `id` | `toLow` | `toHigh`


