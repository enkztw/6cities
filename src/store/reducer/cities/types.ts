enum Types {
  SET_CITY = `SET_CITY`
}

export default Types;
