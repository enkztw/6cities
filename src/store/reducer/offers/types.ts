enum Types {
  SET_OFFER = `SET_OFFER`,
  SET_COMMENTS = `SET_COMMENTS`,
  SET_OFFERS = `SET_OFFERS`,
  SET_FAVORITES = `SET_FAVORITES`,
  SET_NEARBY = `SET_NEARBY`,
  SET_SORTING = `SET_SORTING`
}

export default Types;
