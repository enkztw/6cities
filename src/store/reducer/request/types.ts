enum Types {
  SET_REQUEST = `SET_REQUEST`,
}

export default Types;

