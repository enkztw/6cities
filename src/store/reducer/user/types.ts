enum Types {
  SET_AUTH = `SET_AUTH`
}

export default Types;

