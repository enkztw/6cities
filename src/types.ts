export type property = `apartment` | `room` | `house` | `hotel`;

export type location = {
    latitude: number;
    longitude: number;
    zoom: number;
}

export type id = number
export type offers = Offer[]
export type cityName = `Paris` | `Cologne` | `Brussels` | `Amsterdam` | `Hamburg` | `Dusseldorf`

export interface City {
    location: location;
    name: cityName;
}

export interface User {
    readonly id: id;
    avatarUrl: string;
    isPro: boolean;
    name: string;
    email?: string;
    description?: string;
}

export interface Status {
    isAuthorized: boolean;
}

export interface Review {
    readonly id: id;
    comment: string;
    date: string;
    rating: number;
    user: User;
}

export interface Offer {
    readonly id: id;
    bedrooms: number;
    city: City;
    description: string;
    goods: string[];
    host: User;
    images: string[];
    isFavorite: boolean;
    isPremium: boolean;
    location: location;
    maxAdults: number;
    previewImage: string;
    price: number;
    rating: number;
    title: string;
    type: property;
}

