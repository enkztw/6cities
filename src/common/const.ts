import {city} from './types';

export enum OperationStatus {
  PENDING = `PENDING`,
  SUCCESS = `SUCCESS`,
  FAILED = `FAILED`,
  EMPTY = ``
}

export enum Error {
  UNAUTHORIZED = 401
}

export enum Months {
  January,
  February,
  March,
  April,
  May,
  June,
  July,
  August,
  September,
  October,
  November,
  December
}

export const TIMEOUT = 5000;
export const MAX_IMAGES = 6;
export const BASE_URL = `https://htmlacademy-react-3.appspot.com/six-cities`;

export const AppRoute = {
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  OFFER: `/offer`,
  ROOT: `/`,
};

export const APIRoute = {
  HOTELS: `/hotels`,
  LOGIN: `/login`,
  FAVORITES: `/favorite`,
  getNearbyOffers(id) {
    return `/hotels/${id}/nearby`;
  },
  getComments(id) {
    return `/comments/${id}`;
  }
};

// [width, height] согласно доке leaflet типа Point[number, number]
const PIN_SIZE: [number, number] = [27, 39];
export const PIN = {
  SIZE: PIN_SIZE,
  URL: `img/pin.svg`,
  ACTIVE_URL: `img/pin-active.svg`
};

export const LAYER = {
  URL: `https://{s}.basemaps.cartocdn.com/rastertiles/voyager_labels_under/{z}/{x}/{y}{@2x}.png`
};

export const COMMENT = {
  MIN_LENGTH: 50,
  MAX_LENGTH: 300,
  MAX_COUNT: 10
};

export const cities: city[] = [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];
export const ratings: number[] = [5, 4, 3, 2, 1];
export const sorters = [
  {
    name: `id`,
    title: `Popular`
  },
  {
    name: `toHigh`,
    title: `Price: low to high`
  },
  {
    name: `toLow`,
    title: `Price: high to low`
  },
  {
    name: `rating`,
    title: `Top rated first`
  }];

