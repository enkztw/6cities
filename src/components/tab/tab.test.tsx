import * as React from 'react';
import * as renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';

import {city} from '../../common/types';
import Tab from './tab';

const mockStore = configureStore([]);

describe(`Render <Tab />`, () => {
  it(`<Tab /> should render`, () => {
    const name: city = `Amsterdam`;
    const store = mockStore({
      cities: {
        current: `Paris`
      }
    });

    const tree = renderer.create(
        <Provider store={store}>
          <Tab name={name}/>
        </Provider>
    );

    expect(tree).toMatchSnapshot();
  });

  it(`Active <Tab /> should render`, () => {
    const name: city = `Paris`;
    const store = mockStore({
      cities: {
        current: `Paris`
      }
    });

    const tree = renderer.create(
        <Provider store={store}>
          <Tab name={name}/>
        </Provider>
    );

    expect(tree).toMatchSnapshot();
  });
});
