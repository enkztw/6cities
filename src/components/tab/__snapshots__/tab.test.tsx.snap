// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <Tab /> <Tab /> should render 1`] = `
<li
  className="locations__item"
  onClick={[Function]}
>
  <a
    className="locations__item-link tabs__item"
    href="#"
  >
    <span>
      Amsterdam
    </span>
  </a>
</li>
`;

exports[`Render <Tab /> Active <Tab /> should render 1`] = `
<li
  className="locations__item"
  onClick={[Function]}
>
  <a
    className="locations__item-link tabs__item tabs__item--active"
    href="#"
  >
    <span>
      Paris
    </span>
  </a>
</li>
`;
